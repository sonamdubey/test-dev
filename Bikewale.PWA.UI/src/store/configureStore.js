import { createStore, applyMiddleware} from 'redux'

import thunk from 'redux-thunk'

import rootReducer from '../reducers/rootReducer'

import {fromJS} from 'immutable'

var middleware;
if(process.env.NODE_ENV == 'production') {
	middleware = applyMiddleware(thunk);
}
else {
	var createLogger = require('redux-logger').createLogger;
	middleware = applyMiddleware(thunk,createLogger());  
}

export default function configureStore(initialData) {

	if(initialData == undefined || initialData == null) {
		return createStore(rootReducer,middleware);
	}
	else {
		return createStore(rootReducer,fromJS(initialData),middleware);
		
	}
}
