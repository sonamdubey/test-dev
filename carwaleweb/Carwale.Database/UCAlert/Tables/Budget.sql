CREATE TABLE [UCAlert].[Budget] (
    [BudgetId] SMALLINT NULL,
    [LowerVal] INT      NULL,
    [UpperVal] INT      NULL
);

