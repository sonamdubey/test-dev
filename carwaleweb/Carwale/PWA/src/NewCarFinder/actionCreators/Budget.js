import {
	UPDATE_BUDGET_INPUT_VALUE,
	UPDATE_BUDGET_SLIDER_VALUE
} from '../actionTypes'

import {
	fireInteractiveTracking
} from '../../utils/Analytics'
const updateSliderValue = (values, min, max, userChange) => {
	return {
		type: UPDATE_BUDGET_SLIDER_VALUE,
		values,
		min,
		max,
		userChange
	}
}

const updateBudgetValue = (value) => {
	return {
		type: UPDATE_BUDGET_INPUT_VALUE,
		value
	}
}

const shouldSliderUpdate = (values, slider) => {
	let update = true

	if (slider.values[0] === values[0]) {
		update = false
	}

	return update
}

export const updateBudgetSlider = ({values, min, max, userChange}) => (dispatch, getState) => {
	if (shouldSliderUpdate(values, getState().newCarFinder.budget.slider)) {
		dispatch(updateSliderValue(values, min, max, userChange))
		dispatch(updateBudgetValue(values[0]))
	}
}
