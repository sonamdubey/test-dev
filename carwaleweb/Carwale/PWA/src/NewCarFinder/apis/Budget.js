
import { SUGGESTED_BUDGET_ENDPOINT } from '../constants'

/**
 * This returns a promise to fetch
 * suggested budget in a given city
 * @param {int} cityId
 * @return {Promise<{min, max, suggested}>}
 */
const get = (cityId) => {
    if(cityId){
        return fetch(SUGGESTED_BUDGET_ENDPOINT+"?cityId="+cityId)
        .then(response => {
            if (!response.ok) { throw response }
            return response.json()
        })
    }
    else{
        return Promise.reject(`Invalid parameters: cityId:${cityId}`)
    }
}
const BudgetApi = {
    get
}
export default BudgetApi
