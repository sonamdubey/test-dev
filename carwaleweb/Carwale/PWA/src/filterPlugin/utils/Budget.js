export const formatBudgetTooltipValue = (value) => {
	let budgetValue

	if (value > 9999999) {
		budgetValue = parseFloat((value / 10000000).toFixed(2)) + '+ Cr'
	}
	else if (value <= 200000) {
		budgetValue = 'Upto 2 lakh'
	}
	else {
		budgetValue = parseFloat((value / 100000).toFixed(2)) + ' lakh'
	}

	return budgetValue
}
