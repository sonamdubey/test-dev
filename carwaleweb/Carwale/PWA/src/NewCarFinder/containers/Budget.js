import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import SpeedometerLoader from '../../components/SpeedometerLoader'
import Rheostat from '../../components/Rheostat'
import PitComponent from '../components/BudgetPit'
import {
	updateBudgetSlider
} from '../actionCreators/Budget'
import {
	setCurrentScreen
} from '../actionCreators/Filter'
import BudgetApi from '../apis/Budget'
import {
	formatValueWithComma
} from '../../utils/Common'
import {
	formatBudgetTooltipValue
} from '../utils/Budget'
import {
	getScreenId
} from '../selectors/NCFSelectors'
import { getCityId } from '../../selectors/LocationSelectors'
import {
	fireInteractiveTracking
} from '../../utils/Analytics'
import {CATEGORY_NAME} from '../constants/index';
import { trackCustomData } from '../../utils/cwTrackingPwa';
class Budget extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			isBudgetValid: true,
			budgetErrorText: '',
			oldBudgetAmount: 0,
			isFetching: true
		}
	}
	componentDidMount(){
		const {
			setCurrentScreen,
			screenId,
			cityId,
			updateBudgetSlider,
			slider
		} = this.props
		setCurrentScreen(screenId)
		//fetch suggested budget for city
		if (slider.values[0] <= 0 || !slider.userChange) {
			let {storedBudget} = this.props
			if(storedBudget.length <= 0 || storedBudget[0] <= 0){
				BudgetApi.get(cityId).then(response => {
					Object.keys(response).forEach(key => response[key] = response[key] * 100000)
					const { min, max, suggested } = response
					updateBudgetSlider({ values: [suggested], min, max, userChange: false })
					setTimeout(() => this.setState({ isFetching: false }), 300)
				}).catch(error => {
					updateBudgetSlider({values: [9],min:2,max:100,userChange: false})
					console.log(error)
					this.setState({ isFetching: false })
				})
			}
			else{
				updateBudgetSlider({ values: storedBudget, userChange: false })
				this.setState({ isFetching: false })
			}
		}
		else {
			this.setState({ isFetching: false })
		}
		trackCustomData(CATEGORY_NAME, "BudgetScreenImpression", "NA", false)
	}

	componentWillUnmount() {
		const {
			slider
		} = this.props;
		fireInteractiveTracking(CATEGORY_NAME, "NCF_BudgetFilter", "Final_" + slider.values[0] + "_SliderUsed")
		trackCustomData(CATEGORY_NAME, "BudgetFilterFinalSliderUsed", "value=" + slider.values[0], false)
	}

	componentWillReceiveProps(nextprops){
		const { slider } = nextprops
		this.validateBudget(slider.value)
	}

	handleSliderChange = ({ values }) => {
		const {
			updateBudgetSlider
		} = this.props

		updateBudgetSlider({ values, userChange: true })
	}

	handleSliderDragEnd = () => {
		const{
			slider
		} = this.props;
		fireInteractiveTracking(CATEGORY_NAME,"NCF_BudgetFilter","SliderUsed_"+slider.values[0])
		trackCustomData(CATEGORY_NAME,"BudgetFilterSliderUsed","value="+slider.values[0],false)
	}

	validateBudget = (updatedValue) => {

		const {
			slider
		} = this.props

		if (updatedValue < slider.min) {
			this.setState({
				isBudgetValid: false,
				budgetErrorText: 'Invalid amount'
			})
		}
		else {
			this.setState({
				isBudgetValid: true,
				budgetErrorText: ''
			})
		}
	}

	render() {
		if (this.state.isFetching) {
			return <SpeedometerLoader />
		}
		let {
			slider
		} = this.props

		slider = {
			...slider,
			className: 'budget-rheostat',
			pitComponent: PitComponent,
			pitPoints: [slider.min, slider.max],
			pitPointLabel: formatBudgetTooltipValue,
			snap: true,
			snapOnDragMove: true,
			disableSnapOnClick: true,
			onChange: this.handleSliderChange,
			onSliderDragEnd: this.handleSliderDragEnd,
			handleTooltipLabel: formatBudgetTooltipValue
		}

		return (
			<div className="budget-screen">
				<div className="screen__head">
					<span className="budget-image"></span>
					<h1 className="screen-head__title">{"What's your Budget?"}</h1>
				</div>

				<div className="screen__body">
					<Rheostat
						{...slider}
					/>
				</div>
			</div>
		)
	}
}

const mapStateToProps = (state) => {
	const {
		slider
	} = state.newCarFinder.budget
	const {
		budget: storedBudget
	} = state.filtersScreen.filters
	const screenId = getScreenId(state, 'BudgetFilter')
	const cityId = getCityId(state)
	return {
		slider,
		...cityId,
		screenId,
		storedBudget
	}
}

const mapDispatchToProps = (dispatch, getState) => {
	return {
		updateBudgetSlider: bindActionCreators(updateBudgetSlider, dispatch),
		setCurrentScreen: bindActionCreators(setCurrentScreen, dispatch)
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(Budget)
