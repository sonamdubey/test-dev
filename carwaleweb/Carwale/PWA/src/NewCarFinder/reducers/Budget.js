import { combineReducers } from 'redux'

import {
	UPDATE_BUDGET_SLIDER_VALUE,
	UPDATE_BUDGET_INPUT_VALUE
} from '../actionTypes'

import {
	fireNonInteractiveTracking
} from '../../utils/Analytics'

import snapPoints from '../utils/rheostat/constants/budgetSnapPoints'
import { BUDGET_SLIDER_MIN, BUDGET_SLIDER_MAX } from '../constants/index'
import {CATEGORY_NAME} from '../constants/index'
import { trackCustomData } from '../../utils/cwTrackingPwa'

let initialSliderState = {
	min: BUDGET_SLIDER_MIN,
	max: BUDGET_SLIDER_MAX,
	values: [0],
	userChange:false,
	snapPoints
}

const slider = (state = initialSliderState, action) => {
	switch (action.type) {
		case UPDATE_BUDGET_SLIDER_VALUE:
			if(!action.userChange){
				fireNonInteractiveTracking(CATEGORY_NAME,"NCF_BudgetFilter","Suggested_"+action.values[0])
				trackCustomData(CATEGORY_NAME,"BudgetFilterSuggestion","value="+action.values[0],false)
			}
			return {
				...state,
				values: action.values,
				min: action.min || state.min,
				max: action.max || state.max,
				userChange: action.userChange
			}

		default:
			return state
	}
}


// budget input field
let initialInputState = {
	value: 0
}

const inputBox = (state = initialInputState, action) => {
	switch (action.type) {
		case UPDATE_BUDGET_INPUT_VALUE:
			return {
				...state,
				value: action.value
			}

		default:
			return state
	}
}

export const budget = combineReducers({
	slider,
	inputBox
})
