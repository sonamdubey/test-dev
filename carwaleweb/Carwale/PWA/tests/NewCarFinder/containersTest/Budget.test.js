import React from 'react'
import { Redirect } from 'react-router-dom'
import Budget from '../../../src/NewCarFinder/containers/Budget'
import {
	UPDATE_BUDGET_INPUT_VALUE
} from '../../../src/NewCarFinder/actionTypes'

import thunk from 'redux-thunk'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

jest.mock('../../../src/NewCarFinder/apis/Budget', () => {
    const budgetApiResponse = {
        min:2,
        max:100,
        suggested:39
    }
    return {
        get: jest.fn(( cityId ) => Promise.resolve(budgetApiResponse))
    }
})
describe("Test for budget container", () => {
    let props
    let state = getMockInitialStore()
    let store
    let enzymeWrapper
    const Wrapper = () => {
        if(!enzymeWrapper){
            store = mockStore(state)
            enzymeWrapper = mount(<Provider store={store}><Budget {...props}/></Provider>)
        }
        return enzymeWrapper
    }
    beforeEach(
        () => {
            // props = undefined
            // state = undefined
            store = undefined
            enzymeWrapper = undefined
        })
    it("it should always show default seleced budget",() =>{
        const wrapper = Wrapper()
        expect(wrapper).toMatchSnapshot()

        setTimeout(function() {
            const expectedActions = store.getActions()
            expect(expectedActions).toHaveLength(3)

            expect(expectedActions).toContainEqual({
                type: UPDATE_BUDGET_INPUT_VALUE,
                value:3900000
            })

        }, 0);
    })
})
