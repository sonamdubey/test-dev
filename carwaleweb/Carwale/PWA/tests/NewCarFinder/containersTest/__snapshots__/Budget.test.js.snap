// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for budget container it should always show default seleced budget 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Budget)>
    <Budget
      cityId={1}
      inputBox={
        Object {
          "value": 300000,
        }
      }
      screenId={0}
      setCurrentScreen={[Function]}
      slider={
        Object {
          "max": 10000000,
          "min": 200000,
          "snapPoints": Array [
            200000,
            250000,
            300000,
            350000,
            400000,
            450000,
            500000,
            550000,
            600000,
            650000,
            700000,
            750000,
            800000,
            850000,
            900000,
            950000,
            1000000,
            1050000,
            1100000,
            1150000,
            1200000,
            1250000,
            1300000,
            1400000,
            1500000,
            1600000,
            1700000,
            1800000,
            1900000,
            2000000,
            2100000,
            2200000,
            2300000,
            2400000,
            2500000,
            2600000,
            2700000,
            2800000,
            3000000,
            3200000,
            3400000,
            3500000,
            4000000,
            4500000,
            5000000,
            5500000,
            6000000,
            6500000,
            7000000,
            7500000,
            8000000,
            9000000,
            10000000,
          ],
          "userChange": false,
          "values": Array [
            300000,
          ],
        }
      }
      updateBudgetInput={[Function]}
      updateBudgetSlider={[Function]}
    >
      <SpeedometerLoader>
        <div
          className="speedometer__container"
        >
          <div
            className="speedometer-loader__content"
          >
            <div
              className="speedometer-loader"
            />
            <div
              className="speedometer-needle-circle"
            />
          </div>
        </div>
      </SpeedometerLoader>
    </Budget>
  </Connect(Budget)>
</Provider>
`;
